packages/devtools-evm-hardhat/src/tasks/healthcheck/validate-rpcs.ts

```diff
@@ -0,0 +1,161 @@
+        if (data.result) {
+            return true
+        }
+
+        logger.error(`RPC URL responded with invalid data: ${JSON.stringify(data)}`)
+        return false
+    } catch (error) {
+        logger.error(
+            `Validation failed for RPC URL: ${error instanceof Error ? error.message : 'An unknown error occurred'}`
+        )
+        return false
+    }
+}
+
+const validateWebSocketRpcUrl = async (rpcUrl: string, timeout: number): Promise<boolean> => {
+    return new Promise((resolve) => {
+        const ws = new WebSocket(rpcUrl)
+        const timeoutId = setTimeout(() => {
+            ws.close()
+            logger.error(`WebSocket connection timed out for RPC URL`)
+            resolve(false)
+        }, timeout)
+
+        const cleanup = (success: boolean, message?: string) => {
+            clearTimeout(timeoutId)
+            if (message) {
+                logger.error(message)
+            }
+            ws.close()
+            resolve(success)
+        }
+
+        ws.onopen = () => {
+            const rpcRequest = JSON.stringify({
+                jsonrpc: JSON_RPC,
+                method: 'eth_blockNumber',
+                params: [],
+                id: 1,
+            })
+            ws.send(rpcRequest)
+        }
+
+        ws.onmessage = (event) => {
+            try {
+                const response = JSON.parse(event.data.toString())
+                if (response.jsonrpc === JSON_RPC && !!response.result) {
+                    cleanup(true)
+                } else {
+                    cleanup(false, `Invalid RPC response: ${event.data}`)
+                }
+            } catch (error) {
+                cleanup(false, `Error parsing RPC response: ${error}`)
+            }
+        }
+
+        ws.onerror = (error) => {
+            cleanup(false, `Validation failed: ${error instanceof Error ? error.message : 'An unknown error occurred'}`)
+        }
+
+        ws.onclose = () => {
+            cleanup(false, `Validation failed: WebSocket connection closed unexpectedly`)
+        }
+    })
+}
+
+const validateRpcUrl = async (rpcUrl: string | undefined, timeout: number, networkName: string): Promise<boolean> => {
+    if (!rpcUrl || rpcUrl.trim() === '') {
+        logger.error(`Missing RPC URL for network: ${networkName}`)
+        return false
+    }
+
+    logger.info(`...Validating RPC URL ${rpcUrl} for ${networkName}...`)
+
+    if (rpcUrl.startsWith(HTTP_URL) || rpcUrl.startsWith(HTTPS_URL)) {
+        return await validateHttpsRpcUrl(rpcUrl, timeout)
+    } else if (rpcUrl.startsWith(WS_URL) || rpcUrl.startsWith(WSS_URL)) {
+        return await validateWebSocketRpcUrl(rpcUrl, timeout)
+    }
+
+    logger.error(`Unsupported RPC protocol in network: ${networkName}`)
+
+    return false
+}
+
+const action: ActionType<TaskArguments> = async (taskArgs, hre) => {
+    printLogo()
+
+    const networks = hre.userConfig.networks || {}
+    const networkNames = Object.keys(networks)
```

Comment:
This object might also contain `hardhat` and `localhost` which will in most cases not be available

Reply:
Oh great catch! I will update the logic and cover this scenario with a test as well.